src/app/shared/modules/create-product-modal/create-product-modal.component.ts

```diff
@@ -39,11 +40,38 @@ export class CreateProductModalComponent implements OnDestroy {
           Validators.maxLength(500),
           Validators.pattern(TEXT_REGEX_PATTERN)
         ]
-      ]
+      ],
+      counts: this.fb.array([
+        this.fb.group({
+          location: [''],
+          quantityAvailable: ['', Validators.pattern(NUMERIC_REGEX_PATTERN)],
```

Comment:
Please don't use regexps when they are not necessary. here you can just check that typeof is "number" or use isNaN function


Apply to other places as well.